File: app/src/main/java/uk/co/agfsoft/skyweather/dagger/ApplicationComponent.java
package uk.co.agfsoft.skyweather.dagger;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Component;
import uk.co.agfsoft.skyweather.application.SkyWeatherApplication;



@Component (modules = {ApplicationModule.class})
@Singleton

public interface ApplicationComponent {

    ActivityComponent plus(ActivityModule activityModule);

    void inject(SkyWeatherApplication skyWeatherApplication);
}

File: app/src/main/kotlin/dagger/ApplicationComponent.kt
package dagger

import application.SkyWeatherApplication
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Created by afa28 on 05/02/2018.
 */
@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ApplicationModule::class,
    ViewModule::class,
    BuilderModule::class])

interface ApplicationComponent {
    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: SkyWeatherApplication): Builder

        fun build(): ApplicationComponent
    }

    fun inject(application: SkyWeatherApplication)
}